src/components/ui/OakTextInput/OakTextInput.tsx

```diff
@@ -0,0 +1,196 @@
+import { parseDropShadow } from "@/styles/helpers/parseDropShadow";
+
+type StyledTextInputWrapperProps = {
+  $color: OakCombinedColorToken;
+  $hoverBackground: OakCombinedColorToken;
+  $background: OakCombinedColorToken;
+  $borderColor: OakCombinedColorToken;
+  $focusRingDropShadows: OakDropShadowToken[];
+  $disabledBackgroundColor: OakCombinedColorToken;
+  $readOnlyBorderColor: OakCombinedColorToken;
+  $disabledColor: OakCombinedColorToken;
+  $readOnlyColor: OakCombinedColorToken;
+};
+
+export type OakTextInputProps = {
+  id?: string;
+  type?: "text" | "password" | "number" | "email" | "tel";
+  /**
+   * Disables user input and updates the appearance accordingly.
+   */
+  disabled?: boolean;
+  /**
+   * Makes the input read-only. Preventing the user from changing the value.
+   */
+  readOnly?: boolean;
+  /**
+   * Sets the value. Use this in controlled components;
+   */
+  value?: string;
+  /**
+   * Sets the initial value. Use this for an uncontrolled component;
+   */
+  defaultValue?: string;
+  /**
+   * Used to target the input element in tests.
+   */
+  "data-testid"?: string;
+  placeholder?: string;
+  onChange?: ChangeEventHandler<HTMLInputElement>;
+  /**
+   * Alters the appearance of the input field to indicate whether the input is valid or invalid.
+   */
+  validity?: "valid" | "invalid";
+  /**
+   * Displays an icon at the start of the input.
+   */
+  startEnhancerIconName?: OakIconName;
+  /**
+   * Displays an icon at the end of the input.
+   */
+  endEnhancerIconName?: OakIconName;
+  $iconColor?: OakCombinedColorToken;
+  $validBorderColor?: OakCombinedColorToken;
+  $invalidBorderColor?: OakCombinedColorToken;
+  $validIconColor?: OakCombinedColorToken;
+  $invalidIconColor?: OakCombinedColorToken;
+} & Partial<StyledTextInputWrapperProps> &
+  Pick<InternalTextInputProps, "$width" | "$maxWidth" | "onInitialFocus">;
+
+const StyledTextInputWrapper = styled(OakFlex)<StyledTextInputWrapperProps>`
+  &:hover {
+    cursor: text;
+  }
+
+  &:focus-within:not(:has(input:disabled, input:read-only)) {
+    box-shadow: ${(props) =>
+      props.$focusRingDropShadows
+        .map((dropShadow) => parseDropShadow(dropShadow))
+        .join(",")};
+  }
+
+  background: ${(props) => parseColor(props.$background)};
+
+  @media (hover: hover) {
+    &:hover:not(:focus-within, :has(input:read-only)) {
+      background: ${(props) => parseColor(props.$hoverBackground)};
+    }
+  }
+
+  &:has(input:read-only) {
+    border-color: ${(props) => parseColor(props.$readOnlyBorderColor)};
+    color: ${(props) => parseColor(props.$readOnlyColor)};
+  }
+
+  &:has(input:disabled) {
+    background: ${(props) => parseColor(props.$disabledBackgroundColor)};
+    color: ${(props) => parseColor(props.$disabledColor)};
+  }
+`;
+
+/**
+ * Default input which can be extended to create specialised inputs.
+ */
+export const OakTextInput = ({
+  type = "text",
+  $borderColor = "border-primary",
```

Comment:
if these aren't actually used by styled components then they shouldn't be prefixed with the $ . The reason being that should we later decide to export this as styled-component to allow overrides of styles the $ prefixed props will be consumed before hitting the TSX